lib/backend/k8s/conversion.go

```diff
@@ -209,21 +223,21 @@ func (c converter) podToWorkloadEndpoint(pod *kapiv1.Pod) (*model.KVPair, error)
 	workload := fmt.Sprintf("%s.%s", pod.ObjectMeta.Namespace, pod.ObjectMeta.Name)
 
 	// If the pod doesn't have an IP address yet, then it hasn't gone through CNI.
-	ipNets := []cnet.IPNet{}
-	if c.hasIPAddress(pod) {
-		// Parse the Pod's IP address.
-		_, ipNet, err := cnet.ParseCIDR(fmt.Sprintf("%s/32", pod.Status.PodIP))
-		if err != nil {
-			return nil, err
-		}
-		ipNets = []cnet.IPNet{*ipNet}
+	if !c.hasIPAddress(pod) {
```

Comment:
It feels like the requirement to check if it is a Calico pod is split between the podToWorkloadEndpoint() code and also by some of the code that calls in to podToWorkloadEndpoint() - I just wonder if the interface should be defined that podToWorkloadEndpoint() should not be called for non-Calico pods (and maybe panic if it is).... dunno just thinking out loud.